File: javascript/mix/C06WordcountType2.js
function tallyWords(text) {
    const tally = {};
    const tokens = text.trim().split(/\s+/);
    for (const token of tokens) {
        tally[token] = (tally[token] ?? 0) + 1;
    }
    return tally;
}

module.exports = { tallyWords };

File: cpp/mix/c06_wordcount_type2.cpp
#include <iostream>
#include <sstream>
#include <map>
#include <string>
using namespace std;

map<string, int> tallyWords(const string& text) {
    map<string, int> tally;
    istringstream iss(text);
    string token;

    while (iss >> token) {
        tally[token]++;
    }

    return tally;
}

int main() {
    string text = "apple banana apple orange banana apple";
    map<string, int> result = tallyWords(text);

    for (const auto& [word, count] : result) {
        cout << word << ": " << count << endl;
    }

    return 0;
}
